cli/tests/pcluster/config/test_config_patch.py

```diff
@@ -0,0 +1,266 @@
+from pcluster.config.updatability import UpdatabilityLevel as Updl
+
+defaults = {
+    "master_subnet_id": "subnet-12345678",
+    "compute_subnet_id": "subnet-12345678",
+    "additional_sg": "sg-12345678",
+}
+
+
+def check_patch(src_conf, dst_conf, expected_changes, expected_patch_updatability_level):
+    patch = ConfigPatch(base_config=src_conf, target_config=dst_conf)
+    assert sorted(patch.changes) == sorted(expected_changes)
+    assert patch.updatability_level == expected_patch_updatability_level
+
+
+def test_config_patch(mocker):
+    mocker.patch("pcluster.config.param_types.get_avail_zone", return_value="mocked_avail_zone")
+    src_conf = PclusterConfig()
+    dst_conf = PclusterConfig()
+    # Two new configs must always be equal
+    check_patch(src_conf, dst_conf, [], Updl.ALLOWED)
+
+
+def duplicate_config_file(dst_config_file, test_datadir):
+    # Make a copy of the src template to the target file.
+    # The two resulting PClusterConfig instances will be identical
+    src_config_file_path = os.path.join(str(test_datadir), "pcluster.config.ini")
+    dst_config_file_path = os.path.join(str(test_datadir), dst_config_file)
+    shutil.copy(src_config_file_path, dst_config_file_path)
+
+
+@pytest.mark.parametrize(
+    "section_key, section_label, param_key, src_param_value, dst_param_value, change_updatability_level, "
+    "patch_applicability",
+    [
+        ("vpc", "default", "master_subnet_id", "subnet-12345678", "subnet-1234567a", Updl.DENIED, Updl.DENIED),
+        ("vpc", "default", "additional_sg", "sg-12345678", "sg-1234567a", Updl.ALLOWED, Updl.ALLOWED),
+        ("vpc", "default", "additional_sg", "sg-12345678", "sg-1234567a", Updl.ALLOWED, Updl.ALLOWED),
+    ],
+)
+def test_single_param_change(
+    test_datadir,
+    pcluster_config_reader,
+    mocker,
+    section_key,
+    section_label,
+    param_key,
+    src_param_value,
+    dst_param_value,
+    change_updatability_level,
+    patch_applicability,
+):
+    mocker.patch("pcluster.config.param_types.get_avail_zone", return_value="mocked_avail_zone")
+    dst_config_file = "pcluster.config.dst.ini"
+    duplicate_config_file(dst_config_file, test_datadir)
+
+    src_dict = {}
+    src_dict.update(defaults)
+    src_dict[param_key] = src_param_value
+
+    rendered_config_file = pcluster_config_reader(**src_dict)
+    src_conf = PclusterConfig(config_file=rendered_config_file, fail_on_file_absence=True)
+
+    dst_dict = {}
+    dst_dict.update(defaults)
+    dst_dict[param_key] = dst_param_value
+    rendered_config_file = pcluster_config_reader(dst_config_file, **dst_dict)
+    dst_conf = PclusterConfig(config_file=rendered_config_file,)
+
+    change = Change(
+        section_key, section_label, param_key, src_param_value, dst_param_value, Updatability(change_updatability_level)
+    )
+    check_patch(src_conf, dst_conf, [change], patch_applicability)
+
+
+def test_multiple_param_changes(mocker, pcluster_config_reader, test_datadir):
```

Comment:
I think you can unify the tests: single_param_change and multiple_param_change, by passing the whole cluster content as parameter instead of the single parameter

Reply:
The purpose of the single_param_change test in the future will be to ensure that no parameters will be left without an explicit update policy; I'd keep it separate for this reason.